src/molecules/Quaternion.cpp

```diff
@@ -41,44 +43,48 @@ void Quaternion::multiply_left(const Quaternion& q) {
 }
 
 std::array<double, 3> Quaternion::rotate(const std::array<double, 3>& d) const {
-	std::array<double, 3> drot;
-	double ww = m_qw*m_qw;
-	double xx = m_qx*m_qx;
-	double yy = m_qy*m_qy;
-	double zz = m_qz*m_qz;
-	double wx = m_qw*m_qx;
-	double wy = m_qw*m_qy;
-	double wz = m_qw*m_qz;
-	double xy = m_qx*m_qy;
-	double xz = m_qx*m_qz;
-	double yz = m_qy*m_qz;
-	//          1-2*(yy+zz)
-	drot[0] = (ww+xx-yy-zz)*d[0] + 2.*(xy-wz)*d[1] + 2.*(wy+xz)*d[2];
-	//                            1-2*(xx+zz)
-	drot[1] = 2.*(wz+xy)*d[0] + (ww-xx+yy-zz)*d[1] + 2.*(yz-wx)*d[2];
-	//                                              1-2*(xx+yy)
-	drot[2] = 2.*(xz-wy)*d[0] + 2.*(wx+yz)*d[1] + (ww-xx-yy+zz)*d[2];
+	std::array<double, 3> drot VECTOR3_ZERO;
+	if(VECTOR3_ZERO != d) { // GH#267
```

Comment:
I am not a big fan of commenting the respective PR/issue. This information can also be received by your IDE/extension.